src/api/db/data/20231023110428_remove_unsupported_hook_events.rb

```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class RemoveUnsupportedHookEvents < ActiveRecord::Migration[7.0]
+  def up
+    WorkflowRun.where.not(hook_event: [*SCMWebhookEventValidator::ALLOWED_GITHUB_AND_GITEA_EVENTS, *SCMWebhookEventValidator::ALLOWED_GITLAB_EVENTS]).destroy_all
```

Comment:
```suggestion
    WorkflowRun.where.not(hook_event: [*SCMWebhookEventValidator::ALLOWED_GITHUB_AND_GITEA_EVENTS, *SCMWebhookEventValidator::ALLOWED_GITLAB_EVENTS]).delete_all
```
`delete_all` will trigger the deletion strategy defined in the associations. See: https://github.com/openSUSE/open-build-service/blob/0778664120b4fde2a3e4d43c868afe5aa7e147aa/src/api/app/models/workflow_run.rb#L26-L28

`destroy_all`  won't. Any reason to not trigger them?

Reply:
Yes and we want the callbacks, so we should stick with `destroy_all`.

Reply:
Ups. My fault. Thanks for double checking!